src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2510,8 +2510,13 @@ impl Coordinator {
 
             // Execute the `optimize/mir_to_lir` stage.
             let dataflow_plan = catch_unwind(no_errors, "mir_to_lir", || {
-                Plan::<mz_repr::Timestamp>::finalize_dataflow(dataflow)
-                    .map_err(AdapterError::Internal)
+                Plan::<mz_repr::Timestamp>::finalize_dataflow(
+                    dataflow,
+                    self.catalog()
```

Comment:
I guess without tweaking the `dataflow` construction logic this is not sufficient for triggering the `refine_single_time_dataflow` call.

Reply:
Yes, that is correct. I did not change the setting of `until` here for `dataflow`, because otherwise, we'd show one-shot `SELECT` plans on `EXPLAIN PHYSICAL PLAN` when the feature flag is enabled. I propose that we leave figuring out the right output to produce for `EXPLAIN` as part of https://github.com/MaterializeInc/materialize/issues/18089.